### ISSUE TITLE
wasm2c.js sources/build-routines are missing

### ISSUE THREAD

**Issue by jonassmedegaard** [emscripten-core/emscripten#12688]
script `tools/wasm2c.py` need node module `wasm2c" which is downloaded from <https://registry.npmjs.org/wasm2c/-/wasm2c-1.0.0.tgz> but where is the source for that pre-compiled node module?

<https://www.npmjs.com/package/wasm2c> does not list any source reposotory.

The tarball includes a README.md which says that it "is an Emscripten build of wabt's wasm2c" but what are the build routines?

That included README.md seems to indicate that it can simply be replaced with the regular non-JS wasm2c executable, but that seems to fail (tried using the wabt 1.0.19-1 package in Debian unstable).

Please document how to bootstrap the wasm2c.js, or point to the external project where that is supposed to be documented.

**Comment by sbc100** [comment]
The npm page (https://www.npmjs.com/package/wasm2c) does link to the source no?   It says "This is an Emscripten build of wabt's wasm2" and includes a hyperlink pointing to the source code.  Perhaps there is a more explicit/formal way for a npm package to list its sources?

**Comment by sbc100** [comment]
I agree there should documented steps for how to re-built wasm2c.js from the wabt sources.  I guess the readme of the npm page is probably the best place for that.

In terms of using the non-js version of wasm2c rather than the JS version.. that seems like it should be possible but would probably require a little work.   I imagine emscripten has an implicit dependency on that exact version of wasm2js that is packaged and that recent wabt changes have not be tested with emscripten.

BTW are you planning on actually using wasm2c in production, or is this part of some kind of audit of emscripten dependencies in general?

**Comment by jonassmedegaard** [comment]
Quoting Sam Clegg (2020-11-03 15:49:47)
> BTW are you planning on actually using wasm2c in production, or is this part of some kind of audit of emscripten dependencies in general?

I am packaging emscripten for Debian, where any external tool needs to be built from source - and noticed this issue when inspecting a few of the testsuite failures run against current state of purely-from-Debian resources: https://buildd.debian.org/status/logs.php?pkg=emscripten&arch=all

**Comment by sbc100** [comment]
In that case yes you probably just want the native version of wasm2c.   We should file a separate bug to allow emscripten to use the real (native) wasm2c.

On the other hand you also might want to just skip wasm2c completely.  Its a rather niche feature that probably has very few users as of today.
